lightning/src/ln/outbound_payment.rs

```diff
@@ -2158,7 +2167,7 @@ mod tests {
 		assert!(!outbound_payments.has_pending_payments());
 		assert!(
 			outbound_payments.add_new_awaiting_invoice(
-				payment_id, expiration, Retry::Attempts(0), None, None, OffersContext::Unknown {}
+				payment_id, expiration, Retry::Attempts(0), None, None, Nonce::try_from(&[0u8; 16][..]).unwrap()
```

Comment:
Update the approach so that a nonce is needed to be provided only when an Invoice Request is present in [pr3010.24](https://github.com/shaavan/rust-lightning/commits/pr3010.24)
Thanks for the pointer!